utils/s2n_random.c

```diff
@@ -439,28 +380,69 @@ S2N_RESULT s2n_rand_cleanup(void)
 
 S2N_RESULT s2n_rand_cleanup_thread(void)
 {
-    RESULT_GUARD(s2n_drbg_wipe(&per_thread_private_drbg));
-    RESULT_GUARD(s2n_drbg_wipe(&per_thread_public_drbg));
+    /* Currently, it is only safe for this function to mutate the drbg states
+     * in the per thread rand state. See s2n_defend_against_ube().
+     */
+    RESULT_GUARD(s2n_drbg_wipe(&s2n_per_thread_rand_state.private_drbg));
+    RESULT_GUARD(s2n_drbg_wipe(&s2n_per_thread_rand_state.public_drbg));
+
+    s2n_per_thread_rand_state.drbgs_initialised = false;
 
     return S2N_RESULT_OK;
 }
 
-/*
- * This must only be used for unit tests. Any real use is dangerous and will be overwritten in s2n_defend_if_forked if
- * it is forked. This was added to support known answer tests that use OpenSSL and s2n_get_private_random_data directly.
- */
-S2N_RESULT s2n_set_private_drbg_for_test(struct s2n_drbg drbg)
+static int s2n_rand_urandom_impl(void *ptr, uint32_t size)
```

Comment:
Is it important that you move this method? It makes the diff harder to understand.

It looks like a couple other methods moved around, complicating the diff.